specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-05-01-preview/machineLearningServices.json

```diff
@@ -4996,6 +5703,320 @@
       }
     },
+    //////////////////////////////////////////////////////
+    // Inference Endpoint
+    //////////////////////////////////////////////////////
+    "InferenceEndpointResource": {
+      "type": "object",
+      "description": "Machine Learning Inference Endpoint object wrapped into ARM resource envelope.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "properties": {
+              "description": "Inference Endpoint properties",
+              "properties": {
+                "description": {
+                  "description": "The Endpoint description.",
+                  "type": "string"
+                },
+                "properties": {
+                  "description": "The Endpoint property dictionary. Properties are immutable.",
+                  "type": "object",
+                  "additionalProperties": {
+                    "type": "string"
+                  }
+                },
+                "inferenceEndpointProvisioningState": {
+                  "type": "string",
+                  "enum": [
+                    "Unknown",
+                    "Creating",
+                    "Deleting",
+                    "Succeeded",
+                    "Failed",
+                    "Canceled"
+                  ],
+                  "x-ms-enum": {
+                    "name": "InferenceEndpointProvisioningState",
+                    "modelAsString": true
+                  },
+                  "readOnly": true,
+                  "description": "The current provisioning state of the Inference Deployment resource."
+                },
+                "error": {
+                  "description": "The error details.",
+                  "allOf": [
+                    {
+                      "$ref": "#/definitions/ErrorResponse"
+                    }
+                  ],
+                  "readOnly": true
+                },
+                "authMode":{
+                  "type": "string",
+                  "enum": [
+                    "RBAC",
+                    "AAD",
+                    "Key"
+                  ],
+                  "x-ms-enum": {
+                    "name": "AuthMode",
+                    "modelAsString": true
+                  },
+                  "readOnly": true,
+                  "description": "Authentication used for calling endpoint. Immutable and exclusive."
+                },
+                "uri": {
+                  "description": "The URI for sending inference requests.",
+                  "type": "string",
+                  "readOnly": true
+                }
+              },
+              "x-ms-discriminator-value": "InferenceEndpoint"
+            }
+          }
+        }
+      ]
+    },
+    "InferenceEndpointUpdateParameter": {
+      "description": "The parameters for updating an inference endpoint.",
+      "properties": { }
```

Comment:
is this TODO? can add a TODO comment if so?

Reply:
I left it for conversation and any changes but there is nothing mutable in Endpoint, so we can get rid of PATCH for Endpoint and this object then.